src/main/java/org/zalando/fahrschein/NakadiReader.java

```diff
@@ -121,9 +122,20 @@ private Cursor readCursor(JsonParser jsonParser) throws IOException {
 
         final List<T> events = new ArrayList<>();
 
-        while (jsonParser.nextToken() == JsonToken.START_OBJECT) {
-            final T event = objectReader.readValue(jsonParser, eventClass);
-            events.add(event);
+        if (jsonParser.nextToken() == JsonToken.START_OBJECT) {
+            final Iterator<T> eventIterator = objectReader.readValues(jsonParser, eventClass);
+            while (eventIterator.hasNext()) {
+                try {
+                    events.add(eventIterator.next());
+                } catch (final Exception e) {
```

Comment:
`JsonMappingException` even contains some additional information which might make it possible to handle the error in a sensible way or only ignore certain errors.

On the other hand, if you know where you expect errors, it might be better to just have a more lenient deserializer.